bin/genome-env-next

```diff
@@ -0,0 +1,452 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+
+function usage {
+    sed 's/^\ \{4\}//' <<EOS
+    usage: $THIS [-hsuw] [command args]
+        See \`$THIS -h' for detailed help.
+EOS
+}
+
+function help {
+    sed 's/^\ \{4\}//' <<EOS
+    NAME - $THIS
+
+    DESCRIPTION
+    ===========
+
+    \`$THIS' is used as a wrapper command to setup an environment for \`genome'.  For most use cases no options need to be specified.  For example,
+
+        $THIS genome --help
+
+    If you know you will need to run several commands in succession you can start a new shell by doing,
+
+        $THIS \$SHELL
+
+    OPTIONS
+    =======
+
+    -h    Display this detailed help.
+
+    -M    Disable migration of database.
+
+    -S    Disable configuration of database.
+
+    -s db_snapshot_name
+          The name of the database snapshot to use instead of the default.
+
+    -U    Disable configuration of UR.
+
+    -u ur_repo
+          The path to a UR repo to use instead of the default.
+
+    -v    Enable verbose status messages.
+
+    -W    Disable configuration of Workflow.
+
+    -w workflow_repo
+          The path to a Workflow repo to use instead of the default.
+EOS
+}
+
+function main {
+    local TESTDBSERVER_DB_NAME TESTDBSERVER_DB_HOST TESTDBSERVER_DB_PORT \
+        TESTDBSERVER_DB_USER="genome" \
+        TESTDBSERVER_DB_PASS="mypassword"
+
+    local DISABLE_DB DISABLE_MIGRATIONS DISABLE_UR DISABLE_WF
+    local SNAPSHOT_NAME TOP UR_REPO VERBOSE WF_REPO
+
+    if $GIT rev-parse --is-inside-work-tree 1> /dev/null 2> /dev/null
+    then
+        TOP="$($GIT rev-parse --show-toplevel)"
+    else
+        TOP="$PWD"
+    fi
+
+    export TESTDBSERVER_URL="https://apipe-test-db.gsc.wustl.edu"
+
+    parse_opts "$@"
+    shift $((OPTIND-1))
+
+    rebuild_meta_db
+    setup_test_db "$TOP"
+    setup_env "$TOP"
+
+    log.info ''
+    log.info "Genome Perl: $(genome-perl -e 'printf(qq(%vd\n), $^V)')"
+    log.info "Genome Prove: $(genome-prove --version)"
+    for MODULE in UR Workflow Genome
+    do
+        log.info "${MODULE}: $(genome-perl -M$MODULE -e "print \$INC{q(${MODULE}.pm)}, qq(\n)")"
+    done
+    log.info "Test DB Snapshot: $SNAPSHOT_NAME"
+    log.info "Test DB Name: $TESTDBSERVER_DB_NAME"
+    log.info ''
+
+    if test "$#" -gt 0
+    then
+        log.info "$@"...
+        exec "$@"
+    else
+        if test -n "$BASH"
+        then
+            echo 'Press <Ctrl>-D or type `logout` to return to normal environment.'
+            echo ''
+            if test -z "$GE_ENV_NAME"
+            then
+                GE_ENV_NAME="genome-env"
+            fi
+            unset PROMPT_COMMAND
+            export PS1="(${GE_ENV_NAME}) \w $ "
+            exec bash --norc --noprofile
```

Comment:
A lot of people probably have /gsc/bin/bash as the first bash in their path, but /bin/bash as their login shell. You tested $BASH, why not use that?